### PR TITLE
🚧Feature/edit profile

### PR DESCRIPTION
**Pull request by Corbinj22**
# PR Description
Please include a summary of the change:
This PR includes additions of the editPage component, along with actions to update the user profile on the editPage

Commit Emojis:
:tada: Initial commit
:construction: HTML modifications/updates.
:art: CSS modifications/updates.
:crystal_ball: JS modifications/updates.
:lipstick: hotfix any code.
:bar_chart: new tests
:camera_flash: adding screenshots
:bento: Adding new assets or images
:pencil: DOCS (README.md)

:beetle: If it is a bug or issue fix, describe the issue that was fixed.

If it is a new feature, describe the feature and the iteration of the project spec it addresses.
Allows for user to see and edit their profile email, shipping address, description. 

Note any questions or struggles as well.

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the Turing style guidelines
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: code/web/src/modules/user/Profile.js [hunk 1]
```diff
@@ -35,17 +35,46 @@ const Profile = (props) => (
...
+// user description - P tag displayed on profile
+// dates available - P tag displayed on profile
+// product history - P tag displayed on profile
+// shipping and email address - P tag displayed on profile
+// upcoming deliveries - P tag displayed on profile
+
+// controlled form component to edit all of the above components - navigated to using edit profle button
+// for cohesiveness, stick to the same level and organization of styling used prior (components for H1s, ptags, etc)
+
+
```
Comment (collinkallery) [inline code/web/src/modules/user/Profile.js:77]: We can delete all of these annotations now - they live in the annotations.js file

File: code/web/src/modules/user/api/actions.js [hunk 2]
```diff
@@ -36,7 +37,7 @@ export function login(userCredentials, isLoading = true) {
     return axios.post(routeApi, query({
       operation: 'userLogin',
       variables: userCredentials,
-      fields: ['user {name, email, role}', 'token']
+      fields: ['user {name, email, role, shippingAddress, description, userImage}', 'token']
```
Comment (collinkallery) [inline code/web/src/modules/user/api/actions.js:40]: Data fields requested in our AXIOS query

File: code/web/src/modules/user/Profile.js (hunk 1 above)
Comment (Corbinj22) [reply code/web/src/modules/user/Profile.js:77]: Will do, we can put that up in our next PR

File: code/web/src/modules/user/api/actions.js [hunk 3]
```diff
@@ -117,3 +118,26 @@ export function getGenders() {
...
+    }))
+    .then(respone => {
+      editProfileSetLocalStorageAndCookie(userDetails)
+      dispatch({
+        type: UPDATE_USER,
+        userDetails
+      })
+    })
+  }
+}
```
Comment (collinkallery) [inline code/web/src/modules/user/api/actions.js:138]: Action that mutates a user's data via AXIOS

File: code/web/src/modules/user/api/actions.js [hunk 4]
```diff
@@ -117,3 +118,26 @@ export function getGenders() {
...
+
+const editProfileSetLocalStorageAndCookie = (user) => {
+  const userObject = {email: user.email, name: user.name, role: 'user', shippingAddress: user.shippingAddress}
+  window.localStorage.setItem('user', JSON.stringify(userObj))
+}
```
Comment (collinkallery) [inline code/web/src/modules/user/api/actions.js:143]: Action that saves user information to local storage.

File: code/web/src/modules/user/api/state.js [hunk 5]
```diff
@@ -43,7 +43,13 @@ export default (state = userInitialState, action) => {
         details: null
       }
 
+    case UPDATE_USER:
+      return {
+        ...state,
+        details: action.user
+      }
+
```
Comment (collinkallery) [inline code/web/src/modules/user/api/state.js:51]: Adding in reducer on state file for updating user information.

File: code/web/src/setup/routes/user.js [hunk 6]
```diff
@@ -26,5 +27,10 @@ export default {
     path: '/user/subscriptions',
     component: Subscriptions,
     auth: true
+  },
+
+  editableProfile: {
+    path: '/user/edit-profile',
+    component: EditProfile,
```
Comment (collinkallery) [inline code/web/src/setup/routes/user.js:34]: Route and rendering for EditProfile component.

**Pull request merged**